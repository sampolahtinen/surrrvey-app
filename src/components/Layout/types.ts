export type LayoutProps = {
  children: any;
};
