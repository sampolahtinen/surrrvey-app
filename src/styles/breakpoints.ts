export const breakPoints = {
  phone: "375px",
  tablet: "768px",
  laptop: "1440px"
};
