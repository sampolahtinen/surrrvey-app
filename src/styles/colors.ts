export const colors = {
  white: "white",
  offWhite: "#eeeeee",
  grey: "#323639",
  darkGrey: "#1f2024",
  black: "black",
  red: "#f44336",
  green: "#1ed36f",
  lila: "#6e1fff",
  darkLila: "#4f2848"
};
