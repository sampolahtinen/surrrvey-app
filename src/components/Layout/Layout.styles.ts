import styled from "styled-components";
import { breakPoints } from "../../styles/breakpoints";

export const LayoutContainer = styled.div`
  padding: 10rem;
  width: 100%;
  height: 100vh;

  @media screen and (max-width: ${breakPoints.phone}) {
    padding: 10rem 2.5rem;
  }
`;
