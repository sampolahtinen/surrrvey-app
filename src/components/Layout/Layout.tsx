import React, { Fragment } from "react";
import NavBar from "../NavBar";
import { LayoutContainer } from "./Layout.styles";
import { LayoutProps } from "./types";

const Layout = ({ children }: LayoutProps) => (
  <Fragment>
    <NavBar />
    <LayoutContainer>{children}</LayoutContainer>
  </Fragment>
);

export default Layout;
