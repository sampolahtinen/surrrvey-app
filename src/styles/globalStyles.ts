import { createGlobalStyle } from "styled-components";
import { colors } from "./colors";

export const GlobalStyle = createGlobalStyle`
  @font-face {
  font-family: "GothamRounded";
  src: url("../static/fonts/gothamrounded-medium.otf") format("otf");
  font-display: fallback;
  @import url('https://fonts.googleapis.com/css?family=VT323&display=swap');
  @import url('https://fonts.googleapis.com/css?family=Roboto&display=swap');
  @import url('https://fonts.googleapis.com/css?family=Varela&display=swap');
  }
* {
  padding: 0;
  margin: 0;
  box-sizing: border-box;
  outline: none;
  font-family: "GothamRounded", "Calibre", sans-serif;
}

html {
  font-size: 62.5%;
  overflow-x: hidden;
}

body {
  height: 100%;
  width: 100%;
  font-size: 1.6rem;
  color: ${colors.darkGrey};
  background-attachment: fixed;
  background-color: white;
  font-family: 'Roboto', sans-serif;
  background-image: linear-gradient(to top, #dad4ec 0%, #dad4ec 1%, #f3e7e9 100%);
}

ul,
li {
  list-style: none;
}

a {
  text-decoration: none;
}

input {
  border: 0;
}

@media screen and (max-width: 420px) {
  html {
    font-size: 56.25%; /* 55% of 16px is 9px*/
  }
}

@media screen and (max-width: 768px) {
  html {
    font-size: 56.25%; /* 55% of 16px is 9px*/
  }
}
`;
