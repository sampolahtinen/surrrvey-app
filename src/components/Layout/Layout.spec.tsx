import React from "react";
import { shallow } from "enzyme";
import Layout from "./Layout";
import { LayoutContainer } from "./Layout.styles";

describe("NavBar Component", () => {
  const wrapper = shallow(
    <Layout>
      <div className="test">test</div>
      <div className="test">test</div>
      <div className="test">test</div>
    </Layout>
  );

  it("renders", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("displays children", () => {
    expect(wrapper.find(LayoutContainer).exists()).toBe(true);
    expect(wrapper.find(".test")).toHaveLength(3);
  });
});
