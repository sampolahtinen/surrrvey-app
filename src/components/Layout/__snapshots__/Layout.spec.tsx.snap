// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders 1`] = `
<Fragment>
  <NavBar />
  <styled.div>
    <div
      className="test"
    >
      test
    </div>
    <div
      className="test"
    >
      test
    </div>
    <div
      className="test"
    >
      test
    </div>
  </styled.div>
</Fragment>
`;
